### PR TITLE
Add oauth2 configuration with traefik

### PR DESCRIPTION
**Pull request by antarus**
solution for https://github.com/jhipster/generator-jhipster/issues/9769
and https://github.com/jhipster/generator-jhipster/issues/9770
Add a description for use traefik and oauth2.

### CODE REVIEW

File: pages/traefik.md [hunk 1]
```diff
@@ -61,4 +61,28 @@ As Traefik is using Consul, it will also be useful to check the Consul administr
 
 #### Configure your Base HREF
 
-Before building the gateway's Docker image, you will need to configure the `baseHref` value in `webpack.common.js` to match the gateway base name.  For example, if the gateway base name is `gateway`, the `baseHref` should be `/services/gateway/`.
\ No newline at end of file
+Before building the gateway's Docker image, you will need to configure the `baseHref` value in `webpack.common.js` to match the gateway base name.  For example, if the gateway base name is `gateway`, the `baseHref` should be `/services/gateway/`.
+
+#### Configure for Oauth2
```
Comment (mraible) [inline pages/traefik.md]:
```suggestion
#### Configure for OAuth 2.0
```

File: pages/traefik.md [hunk 2]
```diff
@@ -61,4 +61,28 @@ As Traefik is using Consul, it will also be useful to check the Consul administr
...
+Before building the gateway's Docker image, it is necessary to [Configure your Base HREF](#configure-your-base-href) and update various files.
+
+##### Client
+In `webapp/app/core/login/login.service.ts`, you need to update the `redirectUri` variable to match the gateway base name.
+For example, if the gateway base name is `gateway`, replace `const redirectUri = window.location.origin;` by `const redirectUri = window.location.origin + '/services/gateway/';`
```
Comment (mraible) [inline pages/traefik.md]:
```suggestion
For example, if the gateway base name is `gateway`, replace `const redirectUri = window.location.origin;` with `const redirectUri = window.location.origin + '/services/gateway/';`
```

File: pages/traefik.md [hunk 3]
```diff
@@ -61,4 +61,28 @@ As Traefik is using Consul, it will also be useful to check the Consul administr
...
+
+##### Server
+In `src/main/resources/config/boostrap.yml`, you have to add some tags in consul that will be interpreted as rule in traefik.
+under `- git-branch=${git.branch:}` add
+```
```
Comment (mraible) [inline pages/traefik.md]:
```suggestion
```yaml
```

**Comment by vishal423** [comment]
@antarus, Can we code these changes into the generator instead of documentation?

**Comment by antarus** [comment]
when we generate the gateway, we do not have the information we want to use traefik. We only know it in the docker-compose subgenerator for exemple.

**Comment by antarus** [comment]
change request done.

**Comment by vishal423** [comment]
I was referring to docker-compose sub-gen, but, then it might become a bit complex to update the gateway via this sub-gen.

On a side note:
- Instead of documenting traefik properties under `src/main/resources/config/boostrap.yml`, can those be specified under `generators/docker-compose/templates/central-server-config/application.yml.ejs` and part of generated code?
- I believe, you also need to update `generators/docker-compose/templates/realm-config/jhipster-realm.json.ejs` to allow requests on traefix default port.

**Comment by antarus** [comment]
Good idea, i will try that.

**Comment by antarus** [comment]
I update https://github.com/jhipster/generator-jhipster/pull/9970 with your idea

**Pull request merged**